#! /bin/sh /usr/share/dpatch/dpatch-run
## 50_find_match_stop_at_end_or_beginning_of_buffer.dpatch by Jens Peter Secher <jps@debian.org>

@DPATCH@
diff -urNad sml-mode-4.0/sml-move.el /tmp/dpep.mynKKi/sml-mode-4.0/sml-move.el
--- sml-mode-4.0/sml-move.el	2005-06-12 01:28:09.846013568 +0200
+++ /tmp/dpep.mynKKi/sml-mode-4.0/sml-move.el	2005-06-12 01:34:42.896260848 +0200
@@ -135,6 +135,7 @@
 	(condition-case () (forward-sexp 1) (error (forward-char 1))))
       (setq level
 	    (cond
+	     ((and (eobp) (> level 1)) (error "Unbalanced"))
 	     ((sml-looking-back-at this) (1+ level))
 	     ((sml-looking-back-at match) (1- level))
 	     (t (error "Unbalanced")))))
@@ -150,6 +151,7 @@
 	(condition-case () (backward-sexp 1) (error (backward-char 1))))
       (setq level
 	    (cond
+	     ((and (bobp) (> level 1)) (error "Unbalanced"))
 	     ((looking-at this) (1+ level))
 	     ((looking-at match) (1- level))
 	     (t (error "Unbalanced")))))
